src/track/beatutils.cpp

```diff
@@ -254,52 +247,69 @@ QVector<double> BeatUtils::calculateIronedBeatMap(
     double previousPhaseErrorMean = 0.0;
     auto phaseErrorAvarageCalculator = MovingAvarage(1);
     auto bpmAvarageCalculator = MovingAvarage(1);
-    double meanBpm;
+    double meanBpm = 0.0;
+    double meanBeatLength = 0.0;
+    bool foundFistOutlier = false;
     QVector<double> fixedBeats;
-    while (rightIndex < rawbeats.size()) {
+    while (beatOffset < rawbeats.last() + meanBeatLength) {
         double bpm = 60.0 * sampleRate / (rawbeats[rightIndex] - rawbeats[rightIndex -1]);
         meanBpm = bpmAvarageCalculator(bpm);
         bpmAvarageCalculator.increasePeriod();
         // we do the rounding because we do not want beats at fractional sample position
-        double meanBeatLength = floor(((60.0 * sampleRate) / meanBpm) + 0.5);
+        meanBeatLength = floor(((60.0 * sampleRate) / meanBpm) + 0.5);
```

Comment:
We actually want the beats at fractional position. Only this allows a real const tempo.

Reply:
Right, changed. I remembered read that in some comment somewhere though. Maybe a different context..